src/python/arcor2_arserver/lock.py

```diff
@@ -317,12 +323,12 @@ async def get_root_id(self, obj_id: str) -> str:
             elif obj_id.startswith(SceneObject.uid_prefix()) and obj_id in self.scene.object_ids:
                 # TODO support scene object hierarchy
                 return obj_id
-            elif obj_id in (self.scene.id, self.project.id):
+            elif obj_id in (self.scene.id, self.project.id, LogicItem.START, LogicItem.END):
                 return obj_id
             else:
                 raise Arcor2Exception(f"Error root getter for id '{obj_id}'")
 
-        elif self.scene:
+        elif self.scene and (obj_id in self.scene.object_ids or obj_id == self.scene.id):
```

Comment:
Jenom tip jak to ještě jde alternativně napsat: `elif self.scene and obj_id in self.scene.object_ids | {self.scene.id}`